### DISCUSSION TITLE
Real Time Connection Unubscribing

### DISCUSSION THREAD

**Discussion by kimjust6** [pocketbase/pocketbase#5708] [Q&A]
I apologize if this question was already asked (I tried looking for it in the discussions but couldn't find it)
I am currently extending GO and trying to detect which users are subscribed to a "lobby"

I was able to find which users had subscribed with the hook:
```go
app.OnRealtimeAfterSubscribeRequest().Add(func(e *core.RealtimeSubscribeEvent) error {
		// get client's current subscription
		currentSubscription := e.Subscriptions
		return nil
	})
```

But I am having trouble finding an equivalent when unsubscribing.
There doesn't seem to be an "app.OnRealtimeAfterDisconnectRequest", unless i'm mistaken.

Is it possible to know what collection the user is unsubscribing from? I can only seem to find what the user is already subscribed to:
```go
app.OnRealtimeDisconnectRequest().Add(func(e *core.RealtimeDisconnectEvent) error {

		// get all subs to lobbies
		subs := e.Client.Subscriptions("lobbies")

		for k, v := range e.Client.Subscriptions() {
			log.Println("key:", k) // contains the collection that the user is already subscribed to, but not the current one we are trying to unsubscribe from
			log.Println("val:", v)
		}
return nil
})
```

**Comment by ganigeorgiev** [comment]
It is not clear to me what you are trying to do but the `OnRealtimeDisconnectRequest` is called when the realtime connection disconnects, not just unsubscribe.

A single realtime connection can have 0 or many subscriptions.
Subscribing and unsubscribing trigger the same action - `POST /api/realtime` with the subscription topics.

To detect subscriptions changes BEFORE they are applied you can listen to the [`OnRealtimeBeforeSubscribeRequest`](https://pocketbase.io/docs/go-event-hooks/#onrealtimebeforesubscriberequest) (`e.Subscriptions` will contain the new subscriptions which you can compare against the current `e.Client.Subscriptions()`).

**Comment by kimjust6** [comment]
Thanks for the prompt response.

Essentially, like you have pointed out, I am able to detect what collection the user has subscribed to by using `e.Subscriptions`.
But I am having trouble detecting what collection is being unsubscribed/disconnected from.

In the `OnRealtimeDisconnectRequest`  hook, I would like to be able to tell apart unsubscriptions/disconnections from the `lobby` collection compared to connection disconnects from other collections.

Is this possible?

**Reply by ganigeorgiev** [reply]
I don't think you've read my comment or maybe I'm just bad at explaining things but in all cases I'm not willing to go through another back-and-fort. I'll copy below the important part:

> To detect subscriptions changes BEFORE they are applied you can listen to the [OnRealtimeBeforeSubscribeRequest](https://pocketbase.io/docs/go-event-hooks/#onrealtimebeforesubscriberequest) (e.Subscriptions will contain the new subscriptions which you can compare against the current e.Client.Subscriptions()).

More than this I don't think I can help further.

**Comment by trawor** [comment]
@kimjust6 I managed the problem like this , >=0.23 version(same logic in <0.23):

```
	app.OnRealtimeConnectRequest().BindFunc(func(e *core.RealtimeConnectRequestEvent) error {
		//run next handler at first,
		err := e.Next()

		//at the end of the chain, is the sse connection close
		//clean up old subscriptions
		subs := e.Client.Subscriptions()
		cid := e.Client.Id()
		app.Logger().Debug("Realtime Connect Finished", "clientId", cid, "subs", subs)

		for k, v := range subs {
			fmt.Println("unsub", k, v)
		}

		return err
	})
```

**Reply by kimjust6** [reply]
I'll give this a shot. Thanks.
